### PR TITLE
bs4 fix personal edit

### PR DESCRIPTION
**Pull request by charlienparker**
# PR Checklist

Please check if your PR fulfills the following requirements
- [ ] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [ ] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [ ] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [ ] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [ ] There are no inline styles added
- [ ] There are no inline javascript added
- [ ] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [ ] Code does not use .html_safe
- [ ] All images added/updated have alt text
- [ ] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: 

1. https://www.pivotaltracker.com/story/show/188060838
2. https://www.pivotaltracker.com/story/show/188064194

This PR fixes a bug where the a consumer fields for the primary person appeared below the dependent rows. It also fixes a bug where the cancel button in the person form didn't work. There are also two other unlogged bugs this PR fixes:
- An error caused by recent changes to `show.js.erb` which tried to scroll to elements which exist on the Manage Family page but, critically, _don't_ exist on the Family Info page.
- A UI bug where the Demo Info header didn't appear.

### CODE REVIEW

**Review by ATBull81 — APPROVED** [review]

File: components/financial_assistance/app/views/financial_assistance/shared/_consumer_fields.html.erb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <% if @bs4 %>
   <div id="consumer_fields" class="focus_effect">
+    <h2 class="mt-4"><%= l10n("insured.consumer_roles.demo_info_for") %> <%= @applicant.full_name %></h2>
```
Comment (kristinmerbach) [inline components/financial_assistance/app/views/financial_assistance/shared/_consumer_fields.html.erb:3]: this line needs a check to make sure the applicant is an applicant so it doesn't break the new form. something like below (but with translation)
    <h2 class="mt-4"><%= l10n("insured.consumer_roles.demo_info_for") %> <%= @applicant.class == FinancialAssistance::Applicant ? @applicant.full_name : "this person" %></h2>

**Review by kristinmerbach — APPROVED** [review]
LGTM!

**Comment by kristinmerbach** [comment]
looks good, please resolve the issue flagged first though

File: components/financial_assistance/app/views/financial_assistance/shared/_consumer_fields.html.erb (hunk 1 above)
Comment (charlienparker) [reply components/financial_assistance/app/views/financial_assistance/shared/_consumer_fields.html.erb:3]: Interestingly, the consumer fields partial used on new comes from the main app, `shared/consumer_fields" rendered by `dependent_form`, so this isn't quite a problem. I added this line for robustness, anyway

**Review by kristinmerbach — APPROVED** [review]

**Pull request merged**